dbms/src/Flash/Mpp/ExchangeReceiver.h

```diff
@@ -213,6 +226,11 @@ class ExchangeReceiverBase
     DAGSchema schema;
 
     std::vector<MsgChannelPtr> msg_channels;
+    std::vector<GRPCReceiveQueue<RecvMsgPtr>> grpc_recv_queue;
+    AsyncRequestHandlerWaitQueuePtr async_wait_rewrite_queue;
+
+    std::vector<std::unique_ptr<AsyncRequestHandler<RPCContext, true>>> async_handler_fine_grained_ptrs;
+    std::vector<std::unique_ptr<AsyncRequestHandler<RPCContext, false>>> async_handler_no_fine_grained_ptrs;
```

Comment:
How about
```
class AsyncRequestHandlerBase
{};

template<typename A, bool flag>
class AsyncRequestHandler
{};
```
And then one vector only need here.
```
std::vector<std::unique_ptr<AsyncRequestHandlerBase>> async_handler_ptrs;
```

Reply:
> How about
> 
> ```
> class AsyncRequestHandlerBase
> {};
> 
> template<typename A, bool flag>
> class AsyncRequestHandler
> {};
> ```
> 
> And then one vector only need here.
> 
> ```
> std::vector<std::unique_ptr<AsyncRequestHandlerBase>> async_handler_ptrs;
> ```

okk, It's a good idea.